numba/targets/arraymath.py

```diff
@@ -894,47 +971,42 @@ def check_args(a, b, out):
     check_c_int(context, builder, k)
     check_c_int(context, builder, n)
 
-    fnty = ir.FunctionType(ir.IntType(32),
-                           [ll_char,                       # kind
-                            ll_char_p, ll_char_p,          # transa, transb
-                            intp_t, intp_t, intp_t,        # m, n, k
-                            ll_void_p, ll_void_p, intp_t,  # alpha, a, lda
-                            ll_void_p, intp_t, ll_void_p,  # b, ldb, beta
-                            ll_void_p, intp_t,             # c, ldc
-                           ])
-    fn = builder.module.get_or_insert_function(fnty, name="numba_xxgemm")
-
-    alpha = make_constant_slot(context, builder, dtype, 1.0)
-    beta = make_constant_slot(context, builder, dtype, 0.0)
-
-    trans = context.insert_const_string(builder.module, "t")
-    notrans = context.insert_const_string(builder.module, "n")
-
-    # Since out is C-contiguous, compute a * b = y.T * x.T
-    assert outty.layout == 'C'
-
-    def get_array_param(ty, shapes, data):
-        return (
-                # Transpose if layout different from result's
-                notrans if ty.layout == outty.layout else trans,
-                # Size of the inner dimension in physical array order
-                shapes[1] if ty.layout == 'C' else shapes[0],
-                # The data pointer, unit-less
-                builder.bitcast(data, ll_void_p),
-                )
-
-    transa, lda, data_a = get_array_param(yty, y_shapes, y.data)
-    transb, ldb, data_b = get_array_param(xty, x_shapes, x.data)
-    _, ldc, data_c = get_array_param(outty, out_shapes, out.data)
-
-    kind = get_blas_kind(dtype)
-    kind_val = ir.Constant(ll_char, ord(kind))
-
-    res = builder.call(fn, (kind_val, transa, transb, n, m, k,
-                            builder.bitcast(alpha, ll_void_p), data_a, lda,
-                            data_b, ldb, builder.bitcast(beta, ll_void_p),
-                            data_c, ldc))
-    check_blas_return(context, builder, res)
+    x_data = x.data
+    y_data = y.data
+    out_data = out.data
+
+    # Check whether any of the operands is really a 1-d vector represented
+    # as a (1, k) or (k, 1) 2-d array.  In those cases, it is pessimal
```

Comment:
Fair enough. I guess in general use most people are either in the "vectors are orientation-less data containers" or "vectors are 2D matrix specialisation" category.